### PR TITLE
Feature/openhelpd 124

### PR DESCRIPTION
**Pull request by projectmushroom**
@miltontony @justinvdm @Mitso Please review

_FAQ list page_
- FAQ title corrected
- Batch upload icon changed
- Buttons are now grouped and better aligned
- Translation column added, shows number of available translations
- Parent column removed
- FAQ creation modal added
- Autocomplete dropdown for languages
- Form validation

_FAQ Detail page_
- Left aligned table contents
- Styled labels
- Add Translation button added
- Full language name displayed 
- Removed parent row
- Added a Translation table, rows can be deleted and edited individually
- Create and Edit Translation modal added, Included original FAQ message for reference.
- FAQ edit modal added
- Form validation

screen shot 2016-10-24 at 2 15 48 pm
screen shot 2016-10-26 at 2 46 28 pm
screen shot 2016-10-26 at 2 46 37 pm
screen shot 2016-10-26 at 12 42 26 pm
screen shot 2016-10-26 at 12 42 36 pm
screen shot 2016-10-26 at 12 51 10 pm
screen shot 2016-10-26 at 12 50 16 pm
screen shot 2016-10-26 at 12 50 27 pm
screen shot 2016-10-26 at 12 50 36 pm

### CODE REVIEW

**Comment by justinvdm** [comment]
Looks good :) I think we still need tests for the additions to the controllers and services, and for `FaqModal`.

File: karma/test-controllers.coffee [hunk 1]
```diff
@@ -606,6 +616,147 @@ describe('controllers:', () ->
...
+      $scope = $rootScope.$new()
+      $controller('FaqListController', {$scope: $scope})
+    ))
+  
+    it('init should fetch all FAQs', () ->
+      fetchAllFaqs = spyOnPromise($q, $scope, FaqService, 'fetchAllFaqs')
+  
+      $scope.init()
+  
+      fetchAllFaqs.resolve(test.faq1)
```
Comment (justinvdm) [inline karma/test-controllers.coffee]: Should this be `[test.faq1]`?

File: karma/test-controllers.coffee [hunk 2]
```diff
@@ -606,6 +616,147 @@ describe('controllers:', () ->
...
+      $window.contextData = {faq: test.faq1}
+      $controller('FaqController', {$scope: $scope})
+    ))
+
+    it('init should fetch all FAQs', () ->
+      fetchAllFaqs = spyOnPromise($q, $scope, FaqService, 'fetchAllFaqs')
+
+      $scope.init()
+
+      fetchAllFaqs.resolve(test.faq1)
```
Comment (justinvdm) [inline karma/test-controllers.coffee]: Should this be `[test.faq1]`?

**Comment by justinvdm** [comment]
A few more minor comments, but looks great! 🍰  for all the tests and 🍰 ++ for adding tests for currently untested code that wasn't added in this PR.

File: karma/test-controllers.coffee [hunk 3]
```diff
@@ -606,6 +616,147 @@ describe('controllers:', () ->
...
+      fetchAllFaqs = spyOnPromise($q, $scope, FaqService, 'fetchAllFaqs')
+  
+      $scope.init()
+  
+      fetchAllFaqs.resolve(test.faq1)
+  
+      expect(FaqService.fetchAllFaqs).toHaveBeenCalled()
+      expect($scope.faqs).toEqual(test.faq1)
+      expect($scope.parents).toEqual({})
+    )
```
Comment (projectmushroom) [inline karma/test-controllers.coffee]: You're right I need to test the faq count, will update shortly

**Comment by justinvdm** [comment]
@projectmushroom :+1:!

thanks for your patience with all the review comments :)

**Pull request merged**